ngx-fudis/projects/ngx-fudis/src/lib/directives/grid/grid/grid.directive.spec.ts

```diff
@@ -0,0 +1,199 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { Component } from '@angular/core';
+import { MockComponent } from 'ng-mocks';
+import { getDirective } from '../../../../../utilities/tests/utilities';
+import {
+	FudisGridAlign,
+	FudisGridAlignItems,
+	FudisGridColumnsResponsive,
+	FudisGridGap,
+	FudisGridMarginSide,
+	FudisGridWidth,
+} from '../../../types/grid';
+import { HeadingComponent } from '../../../components/typography/heading/heading.component';
+import { BodyTextComponent } from '../../../components/typography/body-text/body-text.component';
+import { ButtonComponent } from '../../../components/button/button.component';
+import { FudisGridService } from '../../../services/grid/grid.service';
+import { FudisBreakpointService } from '../../../services/breakpoint/breakpoint.service';
+import { GridApiDirective } from '../grid-api/grid-api.directive';
+import { GridDirective } from './grid.directive';
+import { FudisDropdownMenuItemService } from '../../../components/dropdown-menu/dropdown-menu-item/dropdown-menu-item.service';
+import { FudisSpacing } from '../../../types/miscellaneous';
+// import * as utilityFunctions from '../gridUtils';
+
+@Component({
+	selector: 'fudis-mock-grid-directive',
+	template: `<div fudisGrid [align]="align" [alignItemsX]="alignItemsX" [alignItemsY]="alignItemxY" [columns]="3">
+			<fudis-heading [level]="3">I am test heading</fudis-heading>
+			<fudis-body-text>Paragraph text for testing grid directive.</fudis-body-text>
+			<fudis-button [label]="'Test button'" />
+		</div>
+		<div fudisGrid [columns]="2" [marginBottom]="marginBottom" [marginTop]="marginTop" [marginSides]="marginSides">
+			<fudis-heading [level]="4">I am test heading number two</fudis-heading>
+			<fudis-body-text>Paragraph text for testing grid directive.</fudis-body-text>
+			<fudis-button [label]="'Test button number two'" />
+		</div>`,
+})
+class HostComponent {
+	align: FudisGridAlign = 'center';
+
+	alignItemsX: FudisGridAlignItems = 'stretch';
+
+	alignItemsY: FudisGridAlignItems = 'stretch';
+
+	columns: string | number | FudisGridColumnsResponsive = 1;
```

Comment:
Pitäiskö tässä käyttää FudisBreakpointStyleResponsive tyyppiä samoin kuin .ts puolella?

Reply:
The type actually comes from GridApiDirective's `columns` Input which GridDirective extends. GridDirective itself has only protected `_columns` variable.